fili-core/src/main/java/com/yahoo/bard/webservice/async/jobs/stores/JobRowFilter.java

```diff
@@ -0,0 +1,216 @@
+    public JobRowFilter withOperation(FilterOperation operation) {
+        return new JobRowFilter(jobField, operation, values);
+    }
+
+    /**
+     * Construct an JobRowFilter object using the same JobField and FilterOperation as the object on
+     * which this method is called and using the supplied values.
+     *
+     * @param values  A Set of Strings to compare the JobField's value to
+     *
+     * @return  An instance of JobRowFilter created using the supplied values
+     */
+    public JobRowFilter withValues(Set<String> values) {
+        return new JobRowFilter(jobField, operation, values);
+    }
+
+    /**
+     * Extracts the JobField to be examined from the tokenizedQuery.
+     *
+     * @param tokenizedQuery  The tokenized filter expression.
+     *
+     * @return  The JobField to be examined
+     * @throws BadFilterException is the JobField does not exist
+     */
+    private JobField extractJobField(Matcher tokenizedQuery) throws BadFilterException {
+        String fieldName = tokenizedQuery.group(1);
+
+        return Arrays.stream(DefaultJobField.values())
+                .filter(field -> field.getName().equals(fieldName))
+                .findFirst()
+                .orElseThrow(() -> {
+                            LOG.debug(FILTER_JOBFIELD_UNDEFINED.logFormat(fieldName));
+                            return new BadFilterException(FILTER_JOBFIELD_UNDEFINED.format(fieldName));
+                });
+    }
+
+    /**
+     * Extracts the operation to be performed by the ApiJobStore filter query.
+     *
+     * @param tokenizedQuery  The tokenized filter expression..
+     *
+     * @return The operation to be performed by the ApiJobStore filter query.
+     * @throws BadFilterException if the operation does not exist
+     */
+    private FilterOperation extractOperation(Matcher tokenizedQuery) throws BadFilterException {
+        String operationName = tokenizedQuery.group(2);
+        try {
+            return FilterOperation.valueOf(operationName);
+        } catch (IllegalArgumentException ignored) {
+            LOG.debug(FILTER_OPERATOR_INVALID.logFormat(operationName));
+            throw new BadFilterException(FILTER_OPERATOR_INVALID.format(operationName));
+        }
+    }
+
+    /**
+     * Extracts the values to be used in the JobRowFilter query from the query.
+     *
+     * @param tokenizedQuery  The tokenized filter expression..
+     * @param filterQuery  The raw query. Used for logging.
+     *
+     * @return The set of values to be used in the JobRowFilter query.
+     * @throws BadFilterException If the fragment of the query that specifies the values is malformed.
+     */
+    private Set<String> extractValues(Matcher tokenizedQuery, String filterQuery) throws BadFilterException {
+        try {
+            // replaceAll takes care of any leading ['s or trailing ]'s which might mess up the values set.
+            return new LinkedHashSet<>(
+                    FilterTokenizer.split(
```

Comment:
We already have code to handle that case (the ApiFilters handles that already, at least somewhat). There's no reason to _not_ have the filter processing here be at least as good...